interfaces/builtin: Ubuntu Touch unity8-to-lomiri renaming, Stage 1

fredldotme opened the pull request:
Key components comprising the whole Ubuntu Touch experience have been renamed
Lomiri. Among other things DBus namings have changed to com.lomiri and /com/lomiri
respectively. The renaming is part of an ongoing effort to enable the shell
and key system components to be imported into other distributions.

Additionally, on Ubuntu Touch or also Debian/Ubuntu with Lomiri, adapt these interfaces to allow apps to talk to
host-provided services rather than Snaps, since we currently have no plans to replace
our Debian-based services with Snaps.

If we want Ubuntu Touch to make use of snapd more then it needs at least these,
and potentially more, interfaces to work properly.

fredldotme committed:
release: Add release.OnTouch to identify Ubuntu Touch systems

Ubuntu Touch is powered by a similar mechanism to Core for mounting writable paths,
though uses a classic filesystem layout and has other expected behavior from Core,
like most services already being provided by the host.

To enable Touch having more extensive Snap support just add another bool to determine
whether the running system is a Touch image based on a new VARIANT_ID called "touch".

This in turn will provide the basis for further Touch enablement like extrausers support
and Touch-specific permission adjustment.

fredldotme committed:
interfaces/builtin: Adapt location_observe to modern Ubuntu Touch

Key components comprising the whole Ubuntu Touch experience have been renamed
Lomiri. Among other things DBus namings have changed to com.lomiri and /com/lomiri
respectively. The renaming is part of an ongoing effort to enable the shell
and key system components to be imported into other distributions.

Make the "location-observe" interfaces work on Ubuntu Touch by defining the
AppArmor peer label as "unconfined" when run on release.OnTouch, and make it
possible to connect an app Snap's plug to the system-/snapd-provided slot.

fredldotme committed:
interfaces/builtin: Adapt media_hub DBus permits to Lomiri naming

As with location_observe in 4382c4bc6b, rename everything to match the new
DBus API naming convention as part of the Unity8->Lomiri renaming.

fredldotme committed:
interfaces/builtin: Adapt location_control to work on Ubuntu Touch

Same as with location_observe and media_hub.

fredldotme committed:
interfaces/builtin: Append Lomiri-variant of Online Accounts DBus AppArmor rules

Keeps the familiar API around and compatible while matching expectations on the
Lomiri variant of the DBus API.

fredldotme committed:
interfaces: Change maliit slot type to core, use host Maliit OnTouch

Make the maliit interface more like the x11 one which can provide permissions
to either the host-provided socket or an app snap provided one. Specifically,
override the default maliit-server peer label to be unconfined and allow
auto-connecting OnTouch.

Allows to connect to a host-side maliit on Ubuntu Touch 24.04.

zyga reviewed (commented):
Overall this looks good.

Are we sure there are no com.ubuntu.* leftovers that are used on regular systems right now?

fredldotme committed:
interfaces/builtin: Make some Ubuntu Touch interfaces work

Extend screen-inhibit-control, thumbnailer-service, mir & media-hub interfaces to
allow for working permissions to host-provided services. In addition to making those
interfaces be provided by snapd so that one can connect them, use the correct peer
labels for these services when running OnTouch, as all these services run on the host,
not in a confined Snap environment.

This allows lomiri-music-app to play back media over Media Hub.

fredldotme committed:
interfaces: Restore some AutoConnect() returns and set implicitOnClassic for OnTouch

fredldotme committed:
interfaces: Reduce use of release.OnTouch

Detect whether those slots are implicitly connected instead, and only use it
for interfaces which have no mainline (non-hybris) port.

fredldotme committed:
interfaces: Adapt basedeclaration_test for new core interfaces

fredldotme committed:
interfaces: Cleanup unused imports, fix maliit TestConnection

fredldotme committed:
interfaces: Always imply host maliit on release.OnTouch

fredldotme committed:
interfaces: Polic location-observe, media-hub, mir & online-accounts-service

fredldotme committed:
interfaces/builtin: Make media-hub only implicit on Ubuntu Touch

There really is no other consumer of media-hub other than Ubuntu Touch,
it has not been ported to mainline yet.

fredldotme committed:
interfaces/builtin: Remove mediascanner cache access from media-hub

To be declared by the app snap.
